### DISCUSSION TITLE
Add a global search bar in the taskbar like windows 10.

### DISCUSSION THREAD

**Discussion by ritulahkar** [linuxmint/discussions#322] [Ideas]
Add a global search bar in the taskbar, displaying different files, apps and even web results, like windows 10. Even a extention doing the same will be good.

**Comment by rcalixte** [comment]
This sounds like [Cinnamenu](https://cinnamon-spices.linuxmint.com/applets/view/322). You should be able to install it on your system via Applets (also in System Settings).

An excerpt rom the README:
```
Features
 * Fuzzy searching for app names.
 * Filesystem browser and home directory file search.
 * Emoji browser & search.
 * Browser bookmarks and history search
 * Web search suggestions & Wikipedia search.
```

The one area that I'm not sure of is whether there is an applet or extension that supports typing directly into the panel with all of those features. The menu is easy enough to open with a single key press or click though so hopefully it's a non-issue.

**Reply by ritulahkar** [reply]
Thank you for your reply. I do use Cinnamenu. It's has some thing similar. But the things It's lacking is proper internet search, it displays some information from internet but displaying is not good. It should look like as small search results page if needed.

And having the ability to type directly in the taskbar will be good. There is a small extension call internet search. But it ultimately open the browser, even if searching for trivial things like some word meaning. And here also you can't type in the taskbar. But typing in the taskbar is not important than small minimal search results viewer. Which does not exit.
